azurerm/internal/services/videoanalyzer/video_analyzer_resource.go

```diff
@@ -0,0 +1,321 @@
+package videoanalyzer
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/videoanalyzer/mgmt/2021-05-01-preview/videoanalyzer"
+	"github.com/hashicorp/go-azure-helpers/response"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	msiparse "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/msi/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/msi/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/videoanalyzer/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/pluginsdk"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/suppress"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceVideoAnalyzer() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceVideoAnalyzerCreateUpdate,
+		Read:   resourceVideoAnalyzerRead,
+		Update: resourceVideoAnalyzerCreateUpdate,
+		Delete: resourceVideoAnalyzerDelete,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.VideoAnalyzerID(id)
+			return err
+		}),
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:     pluginsdk.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile("^[-a-z0-9]{3,24}$"),
+					"Video Analyzer name must be 3 - 24 characters long, contain only lowercase letters and numbers.",
+				),
+			},
+
+			"location": azure.SchemaLocation(),
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"storage_account": {
+				Type:     pluginsdk.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"id": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: azure.ValidateResourceID,
+						},
+
+						"identity_id": {
```

Comment:
since this has to be a User Assigned Identity ID arguably this'd be clearer as:

```suggestion
						"user_assigned_identity_id": {
```

WDYT?

Reply:
All the other identity lists are called `identity_id`, which is why I went this one, for sake of consistency. If you're not planning on having other resources call this also `user_assigned_identity_id`, I wouldn't change it.

Also, aren't all `identity_id`'s provided user assigned? If not, then it could be that in the future this resource has more options for identity, which would make this a breaking change

Reply:
More explicit is generally preferred and `user_assigned_identity_id` supports that, and is likely the convention we'll aim for across all resources - despite existing resources having to wait until the next major version :)